basic_text_printer.go

```diff
@@ -19,6 +23,11 @@ func (p BasicTextPrinter) WithStyle(style *Style) *BasicTextPrinter {
 	return &p
 }
 
+func (p BasicTextPrinter) WithCustomWriter(writer io.Writer) *BasicTextPrinter {
```

Comment:
Do you think we could just name it `WithWriter`? I think that looks a bit cleaner.
(As every `WithXxx` function is technically "with this custom setting". Also the struct field is called `Writer`, and we normally do `WithStructFieldName`) A simple replace all should do the trick.

Reply:
Good idea! I agree will get that switched over